add patch for HMMER v3.4 to add support for `--seq_limit` option

ThomasHoffmann77 opened the pull request:
(created using `eb --new-pr`)

for #24177

boegel reviewed (changes requested):
@ThomasHoffmann77 Since the patch is only adding an option, I would apply it in the existing easyconfig `HMMER-3.4-gompi-2024a.eb`, rather than introducing a new variant...

ThomasHoffmann77 commented:
> @ThomasHoffmann77 Since the patch is only adding an option, I would apply it in the existing easyconfig `HMMER-3.4-gompi-2024a.eb`, rather than introducing a new variant...

I removed the versionsuffix and renamed the file

Crivella commented:
Test report by @Crivella
**SUCCESS**
Build succeeded for 1 out of 1 (total: 4 mins 0 secs) (1 easyconfigs in total)
crivella-desktop - Linux Ubuntu 22.04.5 LTS (Jammy Jellyfish), x86_64, 13th Gen Intel(R) Core(TM) i9-13900K (skylake), Python 3.11.14
See https://gist.github.com/Crivella/53871503ad379c49e38f43b300bef8bb for a full test report.

Crivella reviewed (approved):
LGTM
